src/python/pants/base/specs.py

```diff
@@ -13,6 +13,8 @@
 from pants.util.dirutil import fast_relpath_optional, recursive_dirname
 from pants.util.frozendict import FrozenDict
 
+SPEC_IGNORE_PREFIX = "!"
```

Comment:
In a separate PR, I'm trying to get `-` working so that you don't have to deal with `''` escaping the `!` in the shell. But that work is independent of this PR -- this PR is all about the class modeling.